admin/c2cgeoportal_admin/schemas/roles.py

```diff
@@ -6,19 +6,27 @@
 )
 from c2cgeoportal_commons.models.main import Role
 
-roles_schema_node = colander.SequenceSchema(
-    GeoFormManyToManySchemaNode(Role),
-    name='restricted_roles',
-    widget=RelationCheckBoxListWidget(
-        Role,
-        'id',
-        'name',
-        order_by='name',
-        edit_url=lambda request, value: request.route_url(
-            'c2cgeoform_item',
-            table='roles',
-            id=value
-        )
-    ),
-    validator=manytomany_validator
-)
+
+def build(name):
+    return colander.SequenceSchema(
+        GeoFormManyToManySchemaNode(Role),
+        name=name,
+        widget=RelationCheckBoxListWidget(
+            Role,
+            'id',
+            'name',
+            order_by='name',
+            edit_url=lambda request, value: request.route_url(
+                'c2cgeoform_item',
+                table='roles',
+                id=value
+            )
+        ),
+        validator=manytomany_validator,
+        missing=colander.drop
+    )
+
+
+roles_schema_node_for_theme = build('restricted_roles')
```

Comment:
I would prefer to call this directly in parent schema contruction in views modules.

Reply:
We were used to clone nodes in view... Is there a difference between scratch building and cloning ?